### PR TITLE
Prevent OgMemberships to be created with the 'non-member' and 'member' role 

### PR DESCRIPTION
**Pull request by dakala**
Continuation of #293 from https://github.com/Gizra/og/pull/147

### CODE REVIEW

File: src/Entity/OgMembership.php [hunk 1]
```diff
@@ -298,8 +297,8 @@ public function preSave(EntityStorageInterface $storage) {
     // This will watch actual empty values and '0'.
     if (!$this->get('uid')->target_id) {
       // Throw a generic logic exception as this will likely get caught in
-      // \Drupal\Core\Entity\Sql\SqlContentEntityStorage::save and turned in an
-      // EntityStorageException anyway.
+      // \Drupal\Core\Entity\Sql\SqlContentEntityStorage::save and turned into
+      // an EntityStorageException anyway.
```
Comment (amitaibu) [inline src/Entity/OgMembership.php:301]: I think the logic to prevent the saving was lost in a recent commit?

**Comment by dakala** [comment]
I've reverted the code that checked the roles from the last good commit: https://github.com/Gizra/og/pull/165/commits/cc20e768e51a85baf5fd753180024a57564afdbd

File: src/Entity/OgMembership.php [hunk 2]
```diff
@@ -319,6 +318,19 @@ public function preSave(EntityStorageInterface $storage) {
...
 
+    // Make sure we don't save non-member or member role with a membership.
+    $roles = [];
+    foreach ($this->getRoles() as $role) {
+      /** @var \Drupal\og\Entity\OgRole $role */
+      if ($role->getName() == OgRoleInterface::ANONYMOUS) {
+        throw new \LogicException('Cannot save an OgMembership with reference to a non-member role.');
+      }
+      elseif ($role->getName() != OgRoleInterface::AUTHENTICATED) {
+        $roles[] = $role;
```
Comment (amitaibu) [inline src/Entity/OgMembership.php]:
``` php
elseif ($role->getName() != OgRoleInterface::AUTHENTICATED) {
  $this->revokeRole($role);
}
```

(so you won't have to `$this->setRoles($roles);`)

**Comment by amitaibu** [comment]
Needs a re-roll. 
Also, can you re-open under the `amitaibu/og` repo - so we can collaborate.

**Comment by dakala** [comment]
Closing in favour of https://github.com/Gizra/og/pull/174

**Pull request closed without merging**